### PR TITLE
Add AutoUpdate Client/Cache implementation

### PR DESCRIPTION
**Pull request by vapopov**
In this PR implemented client and cache layer for the auto update service

Related: https://github.com/gravitational/teleport/pull/39805

### CODE REVIEW

File: api/client/autoupdate/autoupdate.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/gravitational/teleport/api/gen/proto/go/teleport/autoupdate/v1"
+)
+
+// Client is an AutoUpdateService client to manage autoupdate configuration and version.
+type Client struct {
+	grpcClient autoupdate.AutoUpdateServiceClient
+}
+
+// NewClient creates a new AutoUpdateService client.
+func NewClient(grpcClient autoupdate.AutoUpdateServiceClient) *Client {
```
Comment (vapopov) [inline api/client/autoupdate/autoupdate.go:35]: @rosstimothy let me know if this wrapper must be eliminated

File: lib/client/api.go [hunk 2]
```diff
@@ -684,12 +684,6 @@ func RetryWithRelogin(ctx context.Context, tc *TeleportClient, fn func() error,
 		return trace.Wrap(err)
 	}
 
-	if opt.afterLoginHook != nil {
-		if err := opt.afterLoginHook(); err != nil {
-			return trace.Wrap(err)
-		}
-	}
-
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/client/api.go:692]: mistake after merge, this file should be affected, fixing this one

File: lib/auth/authclient/api.go [hunk 3]
```diff
@@ -1176,6 +1183,9 @@ type Cache interface {
 	// DatabaseObjectsGetter defines methods for fetching database objects.
 	services.DatabaseObjectsGetter
 
+	// AutoUpdateServiceGetter defines method for fetching the auto update config and version resources.
+	services.AutoUpdateServiceGetter
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/auth/authclient/api.go:1187]: agree, looks inconsistent, will limit with 2 methods only

File: lib/authz/permissions.go [hunk 4]
```diff
@@ -913,6 +913,8 @@ func roleSpecForProxy(clusterName string) types.RoleSpecV6 {
 				types.NewRule(types.KindSAMLIdPServiceProvider, services.RO()),
 				types.NewRule(types.KindUserGroup, services.RO()),
 				types.NewRule(types.KindClusterMaintenanceConfig, services.RO()),
+				types.NewRule(types.KindAutoUpdateConfig, services.RO()),
+				types.NewRule(types.KindAutoUpdateVersion, services.RO()),
```
[the remark this answers is not in the document]
Comment (vapopov) [reply lib/authz/permissions.go:917]: going to change this one in the next pr, read-only currently minimal requirement for the cache

**Review by rosstimothy — APPROVED** [review]

**Review by sclevine — APPROVED** [review]

**Pull request merged**